core/block.ts

```diff
@@ -2227,28 +2228,33 @@ export class Block implements IASTNodeLocation, IDeletable {
    * @param type The type of the icon to remove from the block.
    * @returns True if an icon with the given type was found, false otherwise.
    */
-  removeIcon(type: string): boolean {
+  removeIcon(type: IconType<any>): boolean {
     if (!this.hasIcon(type)) return false;
     this.getIcon(type)?.dispose();
-    this.icons = this.icons.filter((icon) => icon.getType() !== type);
+    this.icons = this.icons.filter((icon) => !icon.getType().equals(type));
     return true;
   }
 
   /**
    * @returns True if an icon with the given type exists on the block,
    *     false otherwise.
    */
-  hasIcon(type: string): boolean {
-    return this.icons.some((icon) => icon.getType() === type);
+  hasIcon(type: IconType<any>): boolean {
```

Comment:
Can this (here and elsewhere) use `IIcon` instead of `any`? I am not sure if that's legal TS or not.

Reply:
Done!